README.md

```diff
@@ -36,6 +36,72 @@ Cons: Only includes standard libraries - no Debian `-dev` packages.
 
 [MentorGraphics toolchain](http://sourcery.mentor.com/public/gnu_toolchain/arm-none-linux-gnueabi/arm-2014.05-29-arm-none-linux-gnueabi.exe) (formerly known as CodeSourcery).
 
+### Windows 10
+
+Windows 10 supports the Windows Subsystem for Linux which allows us to install and execute the entire compiler toolchain. The steps required to compile native-mode EV3 applications on Windows 10 is as follows:
+
+1. Be sure that your Windows 10 installation has Windows Subsystem for Linux installed. To install it, go to Control Panel --> Programs and Features --> Turn Windows Feature On/Off and check the box next to Windows Subsystem for Linux (Beta).
+
+2. Fire up the bash shell by pressing Start Key, type `bash` and press `Enter`. This will open up Bash on Ubuntu on Windows.
+
+3. Install the ARM compiler by typing `sudo apt-get install arm-linux-gnueabi-gcc`.
+
+4. Use `git clone` to clone this repository to the directory of your choice, e.g.,
+
+```sh
+git clone https://github.com/ddemidov/ev3dev-lang-cpp.git
+```
+
+will clone the repo into a directory called `/ev3dev-lang-cpp.git`.
+
+5. You need to make some changes to the top-level `CMakeLists.txt` file. First, go into the directory
+
+```sh
+cd ev3dev-lang-cpp
+```
+Now, edit the `CMakeLists.txt` file with a text editor of your choice, e.g.,
+
+```sh
+vi CMakeLists.txt
+```
+
+Just after the `project(...)` declaration, set the C/C++ compilers by adding the following lines:
+
+```cmake
+set(CMAKE_CC_COMPILER "arm-linux-gnueabi-gcc")
+set(CMAKE_CXX_COMPILER "arm-linux-gnueabi-g++")
+```
+
+Alternatively, you can set these environment variables during compilation (explained later).
+
+6. You also need the following define, without which the nanosleep-using demos won't compile:
+
+```cmake
+add_definitions(-D_GLIBCXX_USE_NANOSLEEP)
+```
+
+This line can be added immediately after the previous `set(...)` instructions.
+
+7. Now compile your programs and the generated binaries will be ready for EV3. This assumes that you have build tools such as `make` and `cmake` installed - if not, install them with `sudo apt-get install build-essential`. You can perform compilation by invoking the following commands:
+
+```sh
+mkdir build && cd build
+cmake .. -DEV3DEV_PLATFORM=EV3
+make
+```
+
+If your did not set the variables in the `CMakeLists.txt` file, use the following commands instead:
+
+```sh
+mkdir build && cd build
+CC=arm-linux-gnueabi-gcc CXX=arm-linux-gnueabi-g++ cmake ..
+make
+```
+
+8. The `build` directory will now contain folders with binary files ready to be executed on the EV3 brick. The easiest way to copy files is to use a program that supports SFTP, like Filezilla. Remember that, by default, the username of the host system is `robot` and password is `maker`.
```

Comment:
This will be in the WSL home directory by default, in the obscure `AppData/...` directory. Might want to mention that or explain how to cd into your main Windows mountpoint.